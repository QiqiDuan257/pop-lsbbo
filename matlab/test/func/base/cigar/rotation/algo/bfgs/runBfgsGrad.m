clear; clc;

% nohup matlab -nodesktop -nodisplay -nosplash <runBfgsGrad.m >runBfgsGrad.out 2>&1 &

%%
startRun = tic;
randSeed = 20190818; % for reproducibility
funcName = 'popRFuncGradCigar_';
funcDims = [2 10 100 1000 2000 4000 6000 8000 10000]';
% funcDims = [2 10]';
optRes = cell(length(funcDims), 1);
for d = 1 : length(funcDims)
    startRunD = tic;
    funcDim = funcDims(d);
    maxFEval = 1e5 * funcDim;
    rng(randSeed + d);
    x0 = -100 + 100 * rand(funcDim, 1);
    options = optimoptions(@fminunc, ...
        'Algorithm', 'quasi-newton', ...
        'SpecifyObjectiveGradient', true, ...
        'MaxFunctionEvaluations', maxFEval, ...
        'MaxIterations', maxFEval, ...
        'ObjectiveLimit', 1e-10, ...
        'Display', 'off');
    [x, y, exitFlag, output, grad, hessian] = fminunc(...
        funcName, x0, options);
    res = {};
    res.opty = y;
    res.optx = x;
    res.exitFlag = exitFlag;
    res.output = output;
    res.grad = grad;
    res.hessian = hessian;
    optRes{d} = res;
    fprintf('%6d: opty %7.2e + runtime %7.2e.\n', ...
        funcDim, y, toc(startRunD));
end
save('runBfgsGrad.mat', 'optRes', 'funcDims', '-v7.3');
fprintf('Total runtime: %7.5e.\n', toc(startRun));
