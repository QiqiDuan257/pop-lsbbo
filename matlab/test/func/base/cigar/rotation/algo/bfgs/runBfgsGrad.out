
                            < M A T L A B (R) >
                  Copyright 1984-2016 The MathWorks, Inc.
                   R2016b (9.1.0.441655) 64-bit (glnxa64)
                             September 7, 2016

#############################################################################
#             $$$ Enjoy the <pop-lsbbo> pure-MATLAB Library $$$             #
#                        --- version: 0.00.000.1 ---                        #
#############################################################################
     2: opty 5.46e+01 + runtime 9.74e-01.
    10: opty 3.98e+03 + runtime 4.06e-02.
   100: opty 6.16e+03 + runtime 1.29e-01.
  1000: opty 9.61e+03 + runtime 2.31e+00.
  2000: opty 1.16e+01 + runtime 1.44e+01.
  4000: opty 2.39e+03 + runtime 9.25e+01.
  6000: opty 5.53e+03 + runtime 2.51e+02.
  8000: opty 7.62e+03 + runtime 5.49e+02.
 10000: opty 6.35e+02 + runtime 1.03e+03.
Total runtime: 1.99960e+03.
